client/src/pages/SurveyDetail/SurveyDetail.test.jsx

```diff
@@ -1,29 +1,164 @@
+    },
+    {
+      completed: false,
+    },
+  ],
+  questions: [{ position: 1 }],
+  responses: [],
+};
+
+const dummySurveyPublished = {
+  ...dummySurvey,
+  status: 'active',
+  datePublished: Date.now(),
+};
+
+const dummySurveyClosed = {
+  ...dummySurvey,
+  status: 'closed',
+  dateClosed: Date.now(),
+};
 
 describe('Testing the Survey Detail page', () => {
-  it('', (done) => {
-    done();
-    // const { getAllByRole, getByText } = render(
-    //   <Provider store={store}>
-    //     <SurveyDetail />
-    //   </Provider>,
-    // );
+  it('', async () => {
+    // eslint-disable-next-line no-underscore-dangle
+    const mockAxiosGetFirst = jest.spyOn(axios, 'get');
+    mockAxiosGetFirst.mockImplementation((url) => {
+      if (url === '/employees') {
+        return Promise.resolve({ data: dummySurvey.recipients });
+      }
+      return Promise.resolve({ data: dummySurvey });
+    });
+
+    const mockAxiosPatch = jest.spyOn(axios, 'patch');
+    mockAxiosPatch.mockImplementation(() => Promise.resolve({}));
+
+    const { getByText, getByTestId } = render(
+      <BrowserRouter>
+        <Provider store={store}>
+          <SurveyDetail
+            match={{ params: { id: dummySurvey._id } }}
+            surveyId={dummySurvey._id}
+          />
+        </Provider>
+      </BrowserRouter>,
+    );
+
+    const dummySurveyTitle = await waitForElement(() =>
+      getByText(dummySurvey.title),
+    );
+
+    expect(dummySurveyTitle).toBeVisible();
+    expect(getByText(dummySurvey.description)).toBeVisible();
+
+    expect(getByText('Publish Survey')).toBeVisible();
+    expect(getByText('Edit survey')).toBeVisible();
+
+    expect(
+      getByText(`Drafted ${formatDate(dummySurvey.dateCreated)}`),
+    ).toBeVisible();
+    expect(getByText(`Publish pending`)).toBeVisible();
+    expect(getByText(`Closed pending`)).toBeVisible();
+
+    expect(getByText('Recipients')).toBeVisible();
+    expect(
+      getByText('To edit recipient list, select Edit Survey'),
+    ).toBeVisible();
+
+    mockAxiosGetFirst.mockRestore();
+
+    fireEvent.click(getByText('Publish Survey'));
+
+    const mockAxiosGetSecond = jest.spyOn(axios, 'get');
+    mockAxiosGetSecond.mockImplementation((url) => {
+      if (url === '/employees') {
+        return Promise.resolve({ data: dummySurveyPublished.recipients });
+      }
+      return Promise.resolve({ data: dummySurveyPublished });
+    });
+
+    const closeSurveyButton = await waitForElement(() =>
+      getByText('Close Survey'),
+    );
+
+    expect(closeSurveyButton).toBeVisible();
+
+    expect(
+      getByText(`Published ${formatDate(dummySurveyPublished.datePublished)}`),
+    ).toBeVisible();
+
+    expect(getByTestId('survey-detail-stepper')).toBeVisible();
+
+    expect(
+      getByText(
+        `${dummySurveyPublished.responses.length}/${dummySurveyPublished.recipients.length}`,
+      ),
+    ).toBeVisible();
+    expect(getByText('respondents')).toBeVisible();
```

Comment:
You use a lot of `toBeVisible()` in your tests, but I think `toBeInTheDocument` is more informative. Not important to change though.

Reply:
Using to be in the document after getby text feels kind of like saying expect(true).toBeTrue()